### PR TITLE
Apache Kafka Provider Docs fixes

### PR DESCRIPTION
**Pull request by TJaniF**
There were a handful of typos and a formatting issue (duplicated content on the home page leading to two sidebar entries for detailed commits, probably a conflict remnant). Sorry about that! 
Also added a couple of sentences about the Sensors.

cc: @dylanbstorey :)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Nice catch !

File: docs/apache-airflow-providers-apache-kafka/index.rst [hunk 1]
```diff
@@ -54,48 +54,8 @@ Content
...
-
-===================  ==================
-PIP package          Version required
-===================  ==================
-``apache-airflow``   ``>=2.3.0``
-``confluent-kafka``
-===================  ==================
-
-.. include:: ../../airflow/providers/apache/kafka/CHANGELOG.rst
-
```
Comment (eladkal) [inline docs/apache-airflow-providers-apache-kafka/index.rst:95]: Why is this removed?

Comment (potiuk) [reply docs/apache-airflow-providers-apache-kafka/index.rst:95]: It's been duplicated due to conflict resolution likely (see below)

Comment (potiuk) [reply docs/apache-airflow-providers-apache-kafka/index.rst:95]: This part is automatically (re)generated every time you prepare docs, but it was also put accidentally above the comment that marks the end of manually maintained part of the index.

Comment (TJaniF) [reply docs/apache-airflow-providers-apache-kafka/index.rst:95]: Yeah this was my fault, not knowing how it works I had removed the comment, I'm very sorry. 😅

Comment (potiuk) [reply docs/apache-airflow-providers-apache-kafka/index.rst:95]: no no - it's good you removed it :). It should not be there.
